server.js

```diff
@@ -6663,6 +6663,48 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// JetBrains Plugins repository version integration
+camp.route(/^\/jetbrains\/plugin\/(v)\/([^/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var pluginId = match[2];
+  var type = match[1];
+  var format = match[3];
+  var badgeData = getBadgeData('jetbrains plugin', data);
+  var url = 'https://plugins.jetbrains.com/plugins/list?pluginId=' + pluginId;
+
+  request(url, function(err, res, buffer) {
+    if (err || res.statusCode !== 200) {
+      badgeData.text[1] = 'inaccessible';
+      return sendBadge(format, badgeData);
+    }
+
+    xml2js.parseString(buffer.toString(), function (err, data) {
+      if (err) {
+        badgeData.text[1] = 'invalid';
+        return sendBadge(format, badgeData);
+      }
+
+      try {
+        switch (type) {
```

Comment:
Since there is only one case you do not need the `switch` statement at all.

Reply:
@paulmelnikow maybe you didn't see my comments (they are under "show outdated" right know). 
I suggested:
"Most of the code is identical with code from downloads badge (cause we are using the same API resource). Maybe we should have one function handling this?" and "We can have one camp.route with (d|v). switch statement should be put after checking if plugin is not found. Then handle d and v in this switch. You can use "Packagist integration" as a reference."

What do you think about this?

Reply:
I see what you mean. Yes, I agree. Combining this into one route seems like a much better approach.

Reply:
So should I do this?

Reply:
Nevermind. I've done this.